### ISSUE TITLE
SpaceBetweenMethods pass not keeping space around empty constructor with specific pass group

### ISSUE THREAD

**Issue by ferhado** [kokororin/vscode-phpfmt#128]
I've encountered an issue with the `SpaceBetweenMethods` pass in a PHP environment, particularly when used in conjunction with the following group of passes: `StripNewlineAfterClassOpen`, `StripNewlineAfterCurlyOpen`, and `StripNewlineWithinClassBody`. The `SpaceBetweenMethods` pass does not maintain the expected space around an empty constructor.

## Passes Used:
- `SpaceBetweenMethods`
- `StripNewlineAfterClassOpen`
- `StripNewlineAfterCurlyOpen`
- `StripNewlineWithinClassBody`

## Current Behavior:
```php
class ExampleClass {
  public function __construct(private SomeService $service) {}
  public function otherMethod() {
    // do something
  }
}
```

## Expected Behavior:
```php
class ExampleClass {

  public function __construct(private SomeService $service) {}

  public function otherMethod() {
    // do something
  }
}
```

The issue seems to arise when these specific passes are used together. I believe that the `SpaceBetweenMethods` pass should override the other passes to ensure the correct spacing is maintained, especially around empty constructors. This issue is critical for maintaining readability and consistency in the codebase.

Looking forward to any insights or fixes for this problem. Thank you!

**Comment by ferhado** [comment]
I'd like to add a note to the issue: This issue also affects all empty methods, not just constructors, when using the same group of passes.

**Comment by driade** [comment]
Thanks, fixed, please confirm.

**Comment by ferhado** [comment]
Confirmed, issue resolved with v1.1.29. Thank you
